src/runes/runestone.rs

```diff
@@ -68,38 +74,51 @@ impl Runestone {
 
     let Message { mut fields, body } = Message::from_integers(&integers);
 
-    let etching = fields.remove(&TAG_RUNE).map(|rune| Etching {
-      divisibility: fields
-        .remove(&TAG_DIVISIBILITY)
-        .and_then(|divisibility| u8::try_from(divisibility).ok())
-        .and_then(|divisibility| (divisibility <= MAX_DIVISIBILITY).then_some(divisibility))
-        .unwrap_or_default(),
-      limit: fields
-        .remove(&TAG_LIMIT)
-        .and_then(|limit| (limit <= MAX_LIMIT).then_some(limit)),
-      rune: Rune(rune),
-      symbol: fields
-        .remove(&TAG_SYMBOL)
-        .and_then(|symbol| u32::try_from(symbol).ok())
-        .and_then(char::from_u32),
-      term: fields
-        .remove(&TAG_TERM)
-        .and_then(|term| u32::try_from(term).ok()),
-    });
+    let flags = fields.remove(&TAG_FLAGS).unwrap_or_default();
```

Comment:
What is the reason for a tag flag?

Reply:
The thinking here is that we need to indicate that the runestone is an etching, which we could do by having a `TAG_ETCHING` with an empty value. That's a bit inefficient though, since we only need one bit. Instead we introduce `TAG_FLAGS` which is a bitfield of tags, and use the least significant bit to indicate that it's an etching. Later, we can add new flags if we need them.